### PR TITLE
Update Network Overlay to support smarter forced scrolling mechanism

### PR DESCRIPTION
**Pull request by AlanFoster**
When making use of the network inspector on a react-native app, it can be quite annoying that as new requests come in - the network inspector instantly sticks to the bottom.

This PR makes this logic smarter by allowing the user to be scrolled away from the bottom by a particular delta.

## Test Plan

I have split this PR into two commits:

1. Replace ListView with ScrollView
1. Fix the auto-scroll to bottom to only do so when you are intentionally near the bottom of the network inspector

I tested both of these commits individually on both iOS and Android.

## Demo

The network overlay component no longer forcibly scrolls to the bottom unless you are close to the bottom. To let the user know that data _has_ changed we instead flash the scroll bars. Support for this was added under https://github.com/facebook/react-native/pull/15566

Example:

recording

### CODE REVIEW

File: Libraries/Inspector/NetworkOverlay.js [hunk 1]
```diff
@@ -22,12 +22,12 @@ const WebSocketInterceptor = require('WebSocketInterceptor');
 const XHRInterceptor = require('XHRInterceptor');
 
 const LISTVIEW_CELL_HEIGHT = 15;
-const SEPARATOR_THICKNESS = 2;
 
 // Global id for the intercepted XMLHttpRequest objects.
```
[the remark this answers is not in the document]
Comment (AlanFoster) [reply Libraries/Inspector/NetworkOverlay.js]: This is resolved.

**Comment by AlanFoster** [comment]
hey @janicduplessis - I couldn't possibly bother you for a review of this PR? :)
I think it'll really help having a better UX for the network inspector, as well as help to show more examples of FlatList 🤞

**Comment by janicduplessis** [comment]
Hey @AlanFoster thanks for the PR! I'll try to have a look in the next few days :)

File: Libraries/Inspector/NetworkOverlay.js [hunk 2]
```diff
@@ -109,43 +105,48 @@ class NetworkOverlay extends React.Component<Object, {
...
         return;
       }
-      const networkInfo = this._requests[xhrIndex];
-      if (!networkInfo.requestHeaders) {
-        networkInfo.requestHeaders = {};
-      }
-      networkInfo.requestHeaders[header] = value;
-      this._genDetailViewItem(xhrIndex);
+
+      this.setState(function ({ requests }) {
```
Comment (janicduplessis) [inline Libraries/Inspector/NetworkOverlay.js]: Nit: Can you use arrow functions instead, there's a couple other places too.

**Comment by AlanFoster** [comment]
I'll aim to pick this up again, thanks for the review! :+1:

**Comment by AlanFoster** [comment]
Hoping to pick this up again 🤞 

I'll open a separate PR for the work of converting ListView -> FlatList, now that it's officially deprecated

**Pull request closed without merging**